bin/pkg/lock.ml

```diff
@@ -223,15 +239,18 @@ let lock ~version_preference ~lock_dirs_arg =
     ~solver_env_from_current_system
     ~version_preference
     ~lock_dirs
+    ~print_perf_stats
 ;;
 
 let term =
   let+ builder = Common.Builder.term
   and+ version_preference = Version_preference.term
-  and+ lock_dirs_arg = Pkg_common.Lock_dirs_arg.term in
+  and+ lock_dirs_arg = Pkg_common.Lock_dirs_arg.term
+  and+ print_perf_stats = Arg.(value & flag & info [ "print-perf-stats" ]) in
```

Comment:
Do we want to generalize perf improvement to Dune or just to package management? Otherwise, it can be nice to move it to `Common`.

Reply:
IMHO I think for now it's perfectly fine to just improve the locking performance as it is the biggest bottleneck.